dask/distributed/tests/test_client.py

```diff
@@ -111,3 +113,9 @@ def test_register_collections():
         c.close()
         d.close()
 
+
+def test_register_with_scheduler():
+    with scheduler_and_workers() as (s, (a, b)):
+        c = Client(s.address_to_clients)
+        sleep(.2)
```

Comment:
The client also gets a dict of the registered workers and clients when it registers now.